theano/gpuarray/dnn.py

```diff
@@ -1626,11 +1630,12 @@ def apply(self, fgraph):
 
 @register_opt('cudnn', 'fast_compile')
 @op_lifter([SoftmaxGrad])
-def local_softmax_dnn_grad(node, ctx_name):
+#@register_opt2([SoftmaxGrad], 'fast_compile')
+def local_softmax_dnn_grad(op, ctx_name, inputs):
```

Comment:
A test that tests if optimization is not applied on GpuSoftmaxGrad when cudnn is excluded fails with the new optimizer (registeropt2). When I'm disabling the new optimizer, the test passes. Which means, when CUDNN is being used, the new optimizer must know that it should not use the GpuSoftmaxGrad optimization. How to go about this ?
For now, I have disabled the new optimizer